test/gtest/uct/test_md.cc

```diff
@@ -1198,7 +1198,7 @@ UCS_TEST_P(test_cuda, sparse_regions)
 
     /* make CUDA registrations list sparse */
     for (int i = 0; i < count; i++) {
-        if ((i & 1) == 0) {
+        if (((i & 1) == 0) && i) {
```

Comment:
why?

Reply:
because we call reg_mem for ptr[0] on line 1210, so we do not need to free it

Reply:
1. what happens if we keep it as is? did the behavior of cuda_ipc change in this case?
2. if really needed can start the loop from 1 instead of 0